packages/node_modules/@node-red/runtime/lib/flows/index.js

```diff
@@ -365,32 +366,50 @@ async function start(type,diff,muteLog) {
         }
     }
     // Having created or updated all flows, now start them.
-    for (id in activeFlows) {
-        if (activeFlows.hasOwnProperty(id)) {
-            try {
-                activeFlows[id].start(diff);
-
-                // Create a map of node id to flow id and also a subflowInstance lookup map
-                var activeNodes = activeFlows[id].getActiveNodes();
-                Object.keys(activeNodes).forEach(function(nid) {
-                    activeNodesToFlow[nid] = id;
-                });
-            } catch(err) {
-                console.log(err.stack);
+    let startFlows = true
+    try {
+        startFlows = settings.get('flowsRunStateRequested');
+    } catch(err) {
+    }
+    startFlows = (startFlows !== "stop");
+
+    if (startFlows) {
+        for (id in activeFlows) {
+            if (activeFlows.hasOwnProperty(id)) {
+                try {
+                    activeFlows[id].start(diff);
+                    // Create a map of node id to flow id and also a subflowInstance lookup map
+                    var activeNodes = activeFlows[id].getActiveNodes();
+                    Object.keys(activeNodes).forEach(function(nid) {
+                        activeNodesToFlow[nid] = id;
+                    });
+                } catch(err) {
+                    console.log(err.stack);
+                }
             }
         }
+        reallyStarted = true;
+        events.emit("flows:started", {config: activeConfig, type: type, diff: diff});
+        // Deprecated event
+        events.emit("nodes-started");
+    } else {
+        started = false;
     }
-    events.emit("flows:started", {config: activeConfig, type: type, diff: diff});
-    // Deprecated event
-    events.emit("nodes-started");
+
+    const state = {
+        started: reallyStarted,
+        state: reallyStarted ? "started" : "stopped",
```

Comment:
As per other comment - why the need for two properties when they say the same thing just in different types?

Reply:
For human readable state. But yes, I agree this should be rationalised.